pkg/backend/path_creds.go

```diff
@@ -159,7 +159,8 @@ the access token will be available when reading the endpoint.
 
 func pathCreds(b *backend) *framework.Path {
 	return &framework.Path{
-		Pattern: credsPathPrefix + framework.GenericNameWithAtRegex("name") + `$`,
+		// Allow characters not special to urls or shells
+		Pattern: credsPathPrefix + strings.Replace(framework.GenericNameWithAtRegex("name"), "@", "@~!_,:^", 1) + `$`,
```

Comment:
I think I'd actually prefer for us to just define our own utility function here (`credentialNameRegex()` or something). For example if they add another "@" into this function upstream while retaining backward compatibility it could have really unexpected results for us.

Reply:
I'd be willing to do that, if that's what you want.  I looked at the [source code](https://github.com/hashicorp/vault/blob/master/sdk/framework/path.go#L23) and saw that it was a quite complicated pattern and I thought it would be safer to expand the "@" which has to be there given the "WithAt" in the name.  So please confirm that that's what you want.

Reply:
Indeed. I actually find it very confusing to have the dash in the middle of that character class. It must not work this way (in Go?) but I'd expect `[\w-.]` to construct a range from the end(?) of `\w` (`z`) to `.`, which is nothing at all. So I'd prefer to just bring this regex into our code and write it in a way that's easier to understand (i.e., put the `-` at the beginning or end where it belongs). You can also use backticks around the string to not have to double-escape the backslashes.

Reply:
Ok, I made that change and it worked for me in testing.